x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/util/SpatialCoordinateTypes.java

```diff
@@ -61,7 +61,42 @@ public long pointAsLong(double x, double y) {
         }
 
         public SpatialPoint pointAsPoint(Point point) {
-            return new SpatialPoint(point.getX(), point.getY());
+            return makePoint(point.getX(), point.getY());
+        }
+
+        private SpatialPoint makePoint(double x, double y) {
```

Comment:
I don't think we should do this. We should know if we are dealing with geo or cartesian point.